sdk/src/main/java/com/horizen/account/transaction/EthereumTransaction.java

```diff
@@ -512,30 +361,42 @@ public BigInteger getGasLimit() {
     }
 
     @Override
-    public AddressProposition getTo() {
-        return this.to;
+    @JsonIgnore
+    public Optional<AddressProposition> getTo() {
+        return Optional.ofNullable(this.to);
     }
 
     @JsonIgnore
-    public String getToString() {
+    public String getToAddressString() {
         if (this.to != null)
             return BytesUtils.toHexString(this.to.address());
         return "";
     }
 
     @Override
     public synchronized AddressProposition getFrom() {
+        if (this.from == null && this.signature != null) {
+            try {
+                byte[] encodedTransaction = encode(null);
```

Comment:
According the comment posted in the previous CR session:
 I think we don't need to pass signature, but instead a flag to account or not the signature from the Transaction.

Reply:
Ok, done